### PR TITLE
Check local_device_info to avoid crash on reset (uplift to 1.26.x)

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Uplift of #8929
Resolves https://github.com/brave/brave-browser/issues/16066

Pre-approval checklist: 
- [ ] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
post-init CI step is unstable
failed audit-deps https://ci.brave.com/job/pr-brave-browser-pr8929_sync_crash_on_reset_1.26.x-post-init/1/execution/node/145/log/ 

not related to the PR

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.26.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
